sqlglot/dialects/tsql.py

```diff
@@ -218,6 +221,54 @@ def _parse_convert(self, strict):
             # Entails a simple cast without any format requirement
             return self.expression(exp.Cast if strict else exp.TryCast, this=this, to=to)
 
+        def _parse_lateral(self):
```

Comment:
are you sure this is necessary? can we just add a simplified version of this to the generic parser?

Reply:
in general, we try to avoid overriding core methods and putting logic into the generic parser.

the sqlglot parser is supposed to be a superset of as much sql as possible.

Reply:
Would love your input on this. Problems i encountered were:
- CROSS / OUTER are the tokens to identify that its a join. This however should take place in combination of the token of APPLY, as you also have CROSS JOIN. 
- Also the parsing of column specifications in T-SQL is enclosed by parenthesis instead of separated by an alias

If you have any suggestions for me into another direction, im curious to find out.

Reply:
right now you're converting apply to lateral at tokenize time, just keep it as an apply and then you can see when that happens to differentiate.

additionally, we already handle column spec (table alias) parsing i believe

CROSS APPLY (SELECT v, y FROM t) t(v, y)

so for example t(v, y)

if apply, then instead of t AS x, then look for t with table alias

Reply:
I will have a look whether i can integrate it in the generic parser.